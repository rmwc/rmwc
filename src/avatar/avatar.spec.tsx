import React from 'react';
import { render, screen } from '@testing-library/react';
import { Avatar, AvatarCount, AvatarGroup } from './';

describe('Avatar', () => {
  it('renders', () => {
    const { asFragment } = render(<Avatar src="test" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with contain', () => {
    const { asFragment } = render(<Avatar src="test" contain />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders as AvatarGroup', () => {
    const { asFragment } = render(<AvatarGroup />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders as AvatarCount with value of 4', () => {
    const { asFragment } = render(<AvatarCount value={4} />);

    expect(screen.getByText('4')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('handles two names', () => {
    render(<Avatar name="James Friedman" />);

    expect(screen.getByText('JF')).toBeInTheDocument();
  });

  it('handles one name', () => {
    render(<Avatar name="James" />);

    expect(screen.getByText('J')).toBeInTheDocument();
  });
});
