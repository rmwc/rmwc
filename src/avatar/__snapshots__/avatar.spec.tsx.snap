// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders 1`] = `
<DocumentFragment>
  <span
    class="rmwc-icon rmwc-icon--component material-icons rmwc-avatar rmwc-avatar--has-image"
    title=""
  >
    <div
      class="rmwc-avatar__icon"
      style="background-image: url(test); background-size: cover;"
    />
    <div
      class="rmwc-avatar__text"
    >
      <div
        class="rmwc-avatar__text-inner"
      />
    </div>
  </span>
</DocumentFragment>
`;

exports[`Avatar renders as AvatarCount with value of 4 1`] = `
<DocumentFragment>
  <span
    class="rmwc-icon rmwc-icon--component material-icons rmwc-avatar rmwc-avatar--count"
  >
    <div
      class="rmwc-avatar__text"
    >
      <div
        class="rmwc-avatar__text-inner"
      >
        4
      </div>
    </div>
  </span>
</DocumentFragment>
`;

exports[`Avatar renders as AvatarGroup 1`] = `
<DocumentFragment>
  <div
    class="rmwc-avatar-group"
  />
</DocumentFragment>
`;

exports[`Avatar renders with contain 1`] = `
<DocumentFragment>
  <span
    class="rmwc-icon rmwc-icon--component material-icons rmwc-avatar rmwc-avatar--has-image"
    title=""
  >
    <div
      class="rmwc-avatar__icon"
      style="background-image: url(test); background-size: contain;"
    />
    <div
      class="rmwc-avatar__text"
    >
      <div
        class="rmwc-avatar__text-inner"
      />
    </div>
  </span>
</DocumentFragment>
`;
